light_curves/code_src/ztf_functions.py

```diff
@@ -10,316 +11,91 @@
 
-    return ztf_df
-
-
-def transform_lightcurves(ztf_df):
-    """Clean and transform the data into the form expected for a `MultiIndexDFObject`.
-
-    Parameters
-    ----------
-    ztf_df : pd.DataFrame
-        Dataframe of light curves as returned by `load_lightcurves`.
-
-    Returns
-    -------
-    ztf_df : pd.DataFrame
-        The input dataframe, cleaned and transformed.
     """
-    # ztf_df might have more than one light curve per (band + objectid) if the ra/dec is close
-    # to a CCD-quadrant boundary (Sanchez-Saez et al., 2021). keep the one with the most datapoints
-    indexes_to_keep = []
-    for _, singleband_object in ztf_df.groupby(["objectid", "band"]):
-        if len(singleband_object.index) == 1:
-            indexes_to_keep.extend(singleband_object.index)
-        else:
-            npoints = singleband_object["mag"].str.len()
-            npointsmax_object = singleband_object.loc[npoints == npoints.max()]
-            # this may still have more than one light curve if they happen to have the same number
-            # of datapoints (e.g., Yang sample sample_table[7], band 'zr').
-            # arbitrarily pick the one with the min oid.
-            # depending on your science, you may want (e.g.,) the largest timespan instead
-            if len(npointsmax_object.index) == 1:
-                indexes_to_keep.extend(npointsmax_object.index)
-            else:
-                minoid = npointsmax_object.oid.min()
-                indexes_to_keep.extend(npointsmax_object.loc[npointsmax_object.oid == minoid].index)
-    ztf_df = ztf_df.loc[sorted(indexes_to_keep)]
-
-    # store "hmjd" as "time".
-    # note that other light curves in this notebook will have "time" as MJD instead of HMJD.
-    # if your science depends on precise times, this will need to be corrected.
-    ztf_df = ztf_df.rename(columns={"hmjd": "time"})
-
-    # "explode" the data structure into one row per light curve point and set the correct dtypes
-    # note that this operation may require 6+ times the RAM used for ztf_df up to now
-    ztf_df = ztf_df.explode(["time", "mag", "magerr", "catflags"], ignore_index=True)
-    ztf_df = ztf_df.astype({"time": "float", "mag": "float", "magerr": "float", "catflags": "int"})
-
-    # remove data flagged as bad
-    ztf_df = ztf_df.loc[ztf_df["catflags"] < 32768, :]
-
-    # calc flux [https://arxiv.org/pdf/1902.01872.pdf zeropoint corrections already applied]
-    mag = ztf_df["mag"].to_numpy()
-    magerr = ztf_df["magerr"].to_numpy()
-    fluxupper = ((mag - magerr) * u.ABmag).to_value('mJy')
-    fluxlower = ((mag + magerr) * u.ABmag).to_value('mJy')
-    ztf_df["flux"] = (mag * u.ABmag).to_value('mJy')
-    ztf_df["err"] = (fluxupper - fluxlower) / 2
+    # 1) fetch the raw arrays 
+    raw = hats_get_lightcurves(
+        sample_table,
+        object_catalog_path=UPath('s3://irsa-fornax-testdata/ZTF/dr23/objects/hats'),
+        object_margin_cache=None,
```

Comment:
There is now a margin cache available for Objects, so you can do:

```suggestion
        object_margin_cache=UPath('s3://irsa-fornax-testdata/ZTF/dr23/objects/hats_margin_10arcsec'),
```